### PR TITLE
Chore/Missing deps

### PR DESCRIPTION
**Pull request by slavistan**
Includes github issue template, VSCode settings and a missing dev dependency. No version bump as packaged files are not affected.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
         "dist/generated"
     ],
     "scripts": {
-        "test": "npm run build && npx mocha ./dist/tests/**/*.test.js",
+        "test": "npm run build && npx run lint && npx mocha ./dist/tests/**/*.test.js",
```
Comment (jkoenig134) [inline package.json:16]: why do you run lint here?

File: package.json
Comment (jkoenig134) [inline package.json:1]: can you please run `npx sort-package-json`? That should make this more similar to our other package jsons

**Comment by jkoenig134** [comment]
please add a license file including the MIT license

File: package.json (hunk 1 above)
Comment (jkoenig134) [reply package.json:16]: typically we have the following scripts in addition:

```json
{
        "lint": "npm run lint:prettier && npm run lint:eslint && npm run lint:tsc",
        "lint:eslint": "eslint --ext .ts ./src ./test",
        "lint:prettier": "prettier --check .",
        "lint:tsc": "tsc --noEmit && tsc -p test/tsconfig.json --noEmit"
}
```

but you can also do the typechecking in another step

**Review by jkoenig134 — CHANGES_REQUESTED** [review]

**Comment by jkoenig134** [comment]
please also add a test pipeline

File: .prettierignore [hunk 2]
```diff
@@ -0,0 +1 @@
+/demo/_bundle.js
```
Comment (jkoenig134) [inline .prettierignore:1]: you should also ignore `dist` and `generated` here

File: LICENSE [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Copyright 2021 j&s-soft GmbH
```
Comment (jkoenig134) [inline LICENSE:1]: 2023

File: README.md
Comment (jkoenig134) [inline README.md:1]: the README is still not compliant to our readmes. Plase move this readme to README_dev.md and create a README.md that matches our other readmes.

File: .github/workflows/publish-npmjs.yml [hunk 4]
```diff
@@ -1,21 +1,19 @@
...
-  publish-npmjs:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
-        with:
-          node-version: 18.14.2
-          registry-url: https://registry.npmjs.org/
-      - run:
-          npm ci && npm test
```
Comment (jkoenig134) [inline .github/workflows/publish-npmjs.yml:17]: this should also be run in a dedicated test pipeline to make sure that we do not encounter any regression errors

File: .github/workflows/publish.yml [hunk 5]
```diff
@@ -1,19 +1,21 @@
+name: Publish on npmjs.com
```
Comment (jkoenig134) [inline .github/workflows/publish.yml:1]:
```suggestion
name: Publish
```

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -0,0 +1,17 @@
+name: Run tests
+
+on:
+    push:
+        branches-ignore: [main, release/**]
+
+jobs:
+    run-checks:
```
Comment (jkoenig134) [inline .github/workflows/test.yml:8]:
```suggestion
    test:
```

**Review by jkoenig134 — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -14,4 +14,5 @@ jobs:
                   node-version: 18.14.2
                   registry-url: https://registry.npmjs.org/
             - run: npm ci
+            - run: ./.ci/runChecks.sh
```
Comment (jkoenig134) [inline .github/workflows/test.yml:17]: see comment above. this should be an extra job

File: .github/workflows/publish.yml [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+        steps:
+            - uses: actions/checkout@v3
+            - uses: actions/setup-node@v3
+              with:
+                  node-version: 18.14.2
+                  registry-url: https://registry.npmjs.org/
+            - run: npm ci
+            - run: npm run build
+            - run: .ci/runChecks.sh
+            - run: npm publish --access public
```
[the remark this answers is not in the document]
Comment (jkoenig134) [reply .github/workflows/publish.yml]: you missed the part in the p json

[the remark this answers is not in the document]
Comment (slavistan) [reply .github/workflows/publish.yml]: Does `enhanced-publish` accept the `--access public` flag? I'd rather not split up the options between the commandline *and* a configuration file.

[the remark this answers is not in the document]
Comment (jkoenig134) [reply .github/workflows/publish.yml]: it does, but we split it in every project, so we will also split it here.

File: .github/workflows/publish.yml [hunk 9]
```diff
@@ -0,0 +1,21 @@
+name: Publish
+
+on:
+    push:
+        branches: [main, release/**]
+
+jobs:
+    publish-npm:
+        runs-on: ubuntu-latest
+        steps:
```
Comment (jkoenig134) [inline .github/workflows/publish.yml:10]: yaml is kind of                                spacy. Do you use the correct prettier config?

File: package.json [hunk 10]
```diff
@@ -13,16 +13,22 @@
...
-        "lint": "npm run build && npx eslint 'src/**/*.ts' 'tests/**/*.ts'",
-        "demo": "npm run build && npx light-server -s ./demo -w 'src/grammar.peggy, demo/app.js, demo/attributes.js, demo/index.html # cd demo && npm run build && npx webpack --mode production'"
+        "demo": "npm run build && npx light-server -s ./demo -w 'src/grammar.peggy, demo/app.js, demo/attributes.js, demo/index.html # cd demo && npm run build && npx webpack --mode production'",
+        "lint": "npm run lint:prettier && npm run lint:eslint && npm run lint:tsc",
+        "lint:eslint": "eslint ./src ./tests",
+        "lint:prettier": "prettier --check ./src/**/*.ts ./tests/**/*.ts ./demo",
+        "lint:tsc": "tsc --noEmit",
+        "test": "npm run build && npx mocha ./dist/tests/**/*.test.js"
     },
+    "dependencies": {},
```
Comment (jkoenig134) [inline package.json:27]: empty deps object

File: .github/workflows/publish.yml (hunk 9 above)
Comment (slavistan) [reply .github/workflows/publish.yml:10]: I'm using this project's prettier config.

Comment (jkoenig134) [reply .github/workflows/publish.yml:10]: I see, its wrong *this projects* prettier config is wrong :)

```json
{
    "endOfLine": "lf",
    "printWidth": 120,
    "tabWidth": 4,
    "trailingComma": "none",
    "semi": false,
    "arrowParens": "always",
    "overrides": [
        {
            "files": ["*.yaml", "*.yml"],
            "options": {
                "tabWidth": 2
            }
        }
    ]
}
```
should be the correct one

**Review by jkoenig134 — APPROVED** [review]

**Pull request merged**